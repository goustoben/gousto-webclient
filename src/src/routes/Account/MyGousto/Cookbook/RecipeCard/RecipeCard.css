@value Blackberry, TrueBlack from 'styles/colors.css';
@value ScreenSMMin, ScreenMDMin, ScreenLGMin from 'styles/screenSizes.css';
@value sizeXXS, sizeXS, sizeSM, sizeMD from 'styles/typography.css';

.cardContainer {
  color: Blackberry;

  &:hover {
    color: TrueBlack;
  }
}

.image {
  margin-bottom: 10px;
  max-width: 100%;
}

.arrowRight {
  composes: fa-basic-class fa-chevron-right from 'styles/vendor/font-awesome-module.css';
  font-size: sizeXXS;
  @media ScreenSMMin {
    font-size: sizeXS;
  }
  @media ScreenMDMin {
    font-size: sizeXXS;
  }
  @media ScreenLGMin {
    font-size: sizeXS;
  }
}

.title {
  min-height: 65px;
  margin-bottom: 0;
  font-size: sizeSM;
  @media ScreenSMMin {
    min-height: 43px;
    font-size: sizeMD;
  }
  @media ScreenMDMin {
    font-size: sizeSM;
  }
  @media ScreenLGMin {
    font-size: sizeMD;
  }
}
