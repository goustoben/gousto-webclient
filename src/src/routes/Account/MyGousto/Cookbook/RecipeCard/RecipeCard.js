import React from 'react'
import PropTypes from 'prop-types'
import Immutable from 'immutable'
import Image from 'Image'
import { Col } from 'Page/Grid'
import placeholderSrc from 'media/images/product-placeholder.png'
import css from './RecipeCard.css'

const propTypes = {
  link: PropTypes.string,
  images: PropTypes.oneOfType([
    PropTypes.instanceOf(Immutable.List),
    PropTypes.string
  ]).isRequired,
  title: PropTypes.string,
}

const defaultProps = {
  link: null,
  title: null,
}

const maxMediaSize = 400
const placeholderImage = (
  <img className={css.image} src={placeholderSrc} alt="" />
)

const RecipeCard = ({ link, images, title }) => (
  <Col col-xs-6 col-lg-2>
    <a href={link} className={css.cardContainer}>
      <div className={css.cardContent}>
        <Image
          media={images}
          title={title}
          maxMediaSize={maxMediaSize}
          className={css.image}
          placeholder={placeholderImage}
          lazy
        />
        {title && (
          <p className={css.title}>
            {title}&nbsp;
            <span className={css.arrowRight} />
          </p>
        )}
      </div>
    </a>
  </Col>
)

RecipeCard.propTypes = propTypes
RecipeCard.defaultProps = defaultProps

export { RecipeCard }
