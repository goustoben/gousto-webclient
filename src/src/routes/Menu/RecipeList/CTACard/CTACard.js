import React from 'react'
import PropTypes from 'prop-types'
import moment from 'moment'

import Recipe from 'containers/menu/Recipe'

const propTypes = {
  isCurrentCollectionRecommendation: PropTypes.oneOfType([PropTypes.bool, PropTypes.string]),
  thematicName: PropTypes.string,
  deliveryDate: PropTypes.string,
  collectionFilterChange: PropTypes.func
}

const CTACard = ({ isCurrentCollectionRecommendation, thematicName, deliveryDate, collectionFilterChange }) => {
  if (thematicName) {
    // if no date is available, add 3 to the current date for the earliest possible delivery date
    const selectedDate = deliveryDate || moment().add(3, 'days').toISOString()

    return (
      <Recipe
        key={'ctaThematic'}
        view={'ctaThematic'}
        thematicName={thematicName}
        selectedDate={selectedDate}
      />
    )
  }

  if (isCurrentCollectionRecommendation) {
    return (
      <Recipe
        key={'ctaAllRecipe'} 
        view={'ctaAllRecipe'} 
        collectionFilterChange={collectionFilterChange}
      />
    )
  }

  return null
}

CTACard.propTypes = propTypes

export { CTACard }
