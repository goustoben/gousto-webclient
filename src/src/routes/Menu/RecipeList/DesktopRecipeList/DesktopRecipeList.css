.desktopRecipeList {
  display: flex;
}

.recipeColumn {
  flex-basis: 33%;
}

.recipeColumn:nth-child(1),
.recipeColumn:nth-child(2) {
  margin-right: 1rem;
}
