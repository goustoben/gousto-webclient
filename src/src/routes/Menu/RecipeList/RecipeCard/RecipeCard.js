import React from 'react'
import PropTypes from 'prop-types'
import Immutable from 'immutable'

import Recipe from 'containers/menu/Recipe'
import { getFeaturedImage } from 'utils/image'
import { formatRecipeTitle, getFoodBrand } from 'utils/recipe'
import { isRecommendedRecipe } from 'utils/menu'

const getRecipeView = (mobileGridView, isFeatured, isFineDineIn) => {
  if (typeof (window) !== 'undefined' && window.innerWidth < 1025) {
    return 'grid'
  }

  if (mobileGridView) {
    return 'gridSmall'
  }

  if (isFeatured) {
    return 'featured'
  }

  if (isFineDineIn) {
    return 'fineDineIn'
  }

  return 'grid'
}

const RecipeCard = ({
  recipe, index, mobileGridView, showDetailRecipe, isFeatured,

  numPortions, cutoffDate, features, allRecipesList, recipesStore, browserType
}) => {
  if (!recipe) {
    return null
  }

  const recipeId = recipe.get('id')
  const range = getFoodBrand(recipe)
  const isFineDineIn = range.get('slug') === 'fine-dine-in'

  const showShortlistFirstStep = index === 0

  const view = getRecipeView(mobileGridView, isFeatured, isFineDineIn)

  return (
    <Recipe
      id={recipeId}
      position={index}
      title={formatRecipeTitle(recipe.get('title'), recipe.get('boxType', ''), recipe.get('dietType', ''))}
      media={getFeaturedImage(recipe, view, browserType)}
      url={recipe.get('url')}
      useWithin={recipe.get('shelfLifeDays')}
      cookingTime={numPortions === 2 ? recipe.get('cookingTime') : recipe.get('cookingTimeFamily')}
      averageRating={recipe.getIn(['rating', 'average'])}
      ratingCount={recipe.getIn(['rating', 'count'])}
      chef={recipe.get('chef')}
      description={recipe.get('description')}
      availability={recipe.get('availability')}
      equipment={recipe.get('equipment')}
      view={view}
      cutoffDate={cutoffDate}
      onClick={(isViewMoreDetailsClicked = false) => { showDetailRecipe(recipeId, isViewMoreDetailsClicked) }}
      features={features}
      isRecommendedRecipe={isRecommendedRecipe(recipeId, allRecipesList, recipesStore)}
      range={range}
      tasteScore={recipe.getIn(['recommendationData', 'score'])}
      fiveADay={recipe.get('fiveADay')}
      diet={recipe.get('dietType')}
      showShortlistFirstStep={showShortlistFirstStep}
    />
  )
}

RecipeCard.defaultProps = {
  isFeatured: false,
  cutoffDate: ''
}

RecipeCard.propTypes = {
  recipe: PropTypes.instanceOf(Immutable.Map).isRequired,
  index: PropTypes.number.isRequired,
  numPortions: PropTypes.number.isRequired,
  recipesStore: PropTypes.instanceOf(Immutable.Map).isRequired,
  mobileGridView: PropTypes.bool.isRequired,
  features: PropTypes.instanceOf(Immutable.Map).isRequired,
  showDetailRecipe: PropTypes.func.isRequired,
  allRecipesList: PropTypes.instanceOf(Immutable.List).isRequired,
  isFeatured: PropTypes.bool,
  cutoffDate: PropTypes.string,
  browserType: PropTypes.string.isRequired
}

export { RecipeCard }
