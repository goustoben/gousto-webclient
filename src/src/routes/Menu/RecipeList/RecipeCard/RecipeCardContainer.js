import { connect } from 'react-redux'

import { getCutoffDate } from 'routes/Menu/selectors/cutoff'
import { getNumPortions } from 'selectors/basket'

import { RecipeCard } from './RecipeCard'

const mapStateToProps = (state) => ({
  numPortions: getNumPortions(state),
  cutoffDate: getCutoffDate(state),
  features: state.features,
  allRecipesList: state.menuRecipes,
  recipesStore: state.recipes,
  browserType: state.request.get('browser')
})

const RecipeCardContainer = connect(mapStateToProps)(RecipeCard)

export { RecipeCardContainer }
