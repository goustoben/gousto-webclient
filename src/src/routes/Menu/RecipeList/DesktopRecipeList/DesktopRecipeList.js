import React from 'react'
import PropTypes from 'prop-types'
import Immutable from 'immutable'

import { arrayToColumns } from '../arrayToColumns'
import { RecipeCardContainer } from '../RecipeCard'
import { CTACard } from '../CTACard'

import css from './DesktopRecipeList.css'

const DesktopRecipeList = ({
  recipes, thematicName, isCurrentCollectionRecommendation, deliveryDate
}) => {
  // eslint-disable-next-line react/prop-types
  const createRecipeCard = ({ index, value }) => (
    <RecipeCardContainer
      key={`${index}-${value.recipe.get('id')}`} // eslint-disable-line react/prop-types
      recipeId={value.recipe.get('id')} // eslint-disable-line react/prop-types
      originalId={value.originalId} // eslint-disable-line react/prop-types
      index={index}
    />
  )

  const recipeArr = recipes.toArray()

  const [left, middle, right] = arrayToColumns(recipeArr, 3, 0)

  return (
    <div className={css.desktopRecipeList}>
      <div className={css.recipeColumn}>
        {left.map(createRecipeCard)}
      </div>
      <div className={css.recipeColumn}>
        {middle.map(createRecipeCard)}

        <CTACard
          isCurrentCollectionRecommendation={isCurrentCollectionRecommendation}
          thematicName={thematicName}
          deliveryDate={deliveryDate}
        />
      </div>
      <div className={css.recipeColumn}>
        {right.map(createRecipeCard)}
      </div>
    </div>
  )
}

DesktopRecipeList.propTypes = {
  recipes: PropTypes.instanceOf(Immutable.List).isRequired,
  isCurrentCollectionRecommendation: PropTypes.oneOfType([PropTypes.bool, PropTypes.string]),
  thematicName: PropTypes.string,
  deliveryDate: PropTypes.string,
}

DesktopRecipeList.defaultProps = {
  thematicName: null,
  deliveryDate: null,
  isCurrentCollectionRecommendation: false,
}

export { DesktopRecipeList }
