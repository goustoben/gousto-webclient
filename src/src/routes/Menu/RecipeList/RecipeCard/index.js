import { RecipeCardContainer } from './RecipeCardContainer'

export { RecipeCardContainer as RecipeCard }
