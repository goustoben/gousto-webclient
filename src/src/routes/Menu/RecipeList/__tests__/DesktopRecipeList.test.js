import React from 'react'
import { shallow } from 'enzyme'
import Immutable from 'immutable'

import { DesktopRecipeList } from '../DesktopRecipeList'
import { CTACard } from '../CTACard'
import { RecipeCardContainer } from '../RecipeCard'

describe('DesktopRecipeList', () => {
  describe('when given 4 recipes', () => {
    test('should render 4 recipes', () => {
      const wrapper = shallow(
        <DesktopRecipeList
          recipes={Immutable.fromJS([{}, {}, {}, {}])}
          showDetailRecipe={() => null}
          thematicName={null}
          isCurrentCollectionRecommendation={false}
          deliveryDate={null}
          collectionFilterChange={() => null}
        />
      )

      expect(wrapper.find(RecipeCardContainer)).toHaveLength(4)
    })
  })

  test('should render a CTACard', () => {
    const wrapper = shallow(
      <DesktopRecipeList
        recipes={Immutable.fromJS([])}
      />
    )

    expect(wrapper.find(CTACard)).toHaveLength(1)
  })
})
