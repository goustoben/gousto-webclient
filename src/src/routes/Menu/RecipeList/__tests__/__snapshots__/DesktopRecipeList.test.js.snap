// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DesktopRecipeList when given 4 recipes should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <DesktopRecipeList
    collectionFilterChange={[Function]}
    deliveryDate={null}
    isCurrentCollectionRecommendation={false}
    mobileGridView={false}
    recipes={
      Immutable.List [
        Immutable.Map {},
        Immutable.Map {},
        Immutable.Map {},
        Immutable.Map {},
      ]
    }
    showDetailRecipe={[Function]}
    thematicName={null}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="mainColumnLeft"
        >
          <div
            className="featured"
          >
            <Connect(RecipeCard)
              index={0}
              isFeatured={true}
              mobileGridView={false}
              recipe={Immutable.Map {}}
              showDetailRecipe={[Function]}
            />
          </div>
          <div
            className="containerBelowFeatured"
          >
            <div
              className="columnBelowFeatured"
            >
              <Connect(RecipeCard)
                index={2}
                mobileGridView={false}
                recipe={Immutable.Map {}}
                showDetailRecipe={[Function]}
              />
            </div>
            <div
              className="columnBelowFeatured"
            >
              <Connect(RecipeCard)
                index={3}
                mobileGridView={false}
                recipe={Immutable.Map {}}
                showDetailRecipe={[Function]}
              />
              <CTACard
                collectionFilterChange={[Function]}
                deliveryDate={null}
                isCurrentCollectionRecommendation={false}
                thematicName={null}
              />
            </div>
          </div>
        </div>,
        <div
          className="mainColumnRight"
        >
          <Connect(RecipeCard)
            index={1}
            mobileGridView={false}
            recipe={Immutable.Map {}}
            showDetailRecipe={[Function]}
          />
        </div>,
      ],
      "className": "desktopRecipeList",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="featured"
            >
              <Connect(RecipeCard)
                index={0}
                isFeatured={true}
                mobileGridView={false}
                recipe={Immutable.Map {}}
                showDetailRecipe={[Function]}
              />
            </div>,
            <div
              className="containerBelowFeatured"
            >
              <div
                className="columnBelowFeatured"
              >
                <Connect(RecipeCard)
                  index={2}
                  mobileGridView={false}
                  recipe={Immutable.Map {}}
                  showDetailRecipe={[Function]}
                />
              </div>
              <div
                className="columnBelowFeatured"
              >
                <Connect(RecipeCard)
                  index={3}
                  mobileGridView={false}
                  recipe={Immutable.Map {}}
                  showDetailRecipe={[Function]}
                />
                <CTACard
                  collectionFilterChange={[Function]}
                  deliveryDate={null}
                  isCurrentCollectionRecommendation={false}
                  thematicName={null}
                />
              </div>
            </div>,
          ],
          "className": "mainColumnLeft",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <Connect(RecipeCard)
                index={0}
                isFeatured={true}
                mobileGridView={false}
                recipe={Immutable.Map {}}
                showDetailRecipe={[Function]}
              />,
              "className": "featured",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "index": 0,
                "isFeatured": true,
                "mobileGridView": false,
                "recipe": Immutable.Map {},
                "showDetailRecipe": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <div
                  className="columnBelowFeatured"
                >
                  <Connect(RecipeCard)
                    index={2}
                    mobileGridView={false}
                    recipe={Immutable.Map {}}
                    showDetailRecipe={[Function]}
                  />
                </div>,
                <div
                  className="columnBelowFeatured"
                >
                  <Connect(RecipeCard)
                    index={3}
                    mobileGridView={false}
                    recipe={Immutable.Map {}}
                    showDetailRecipe={[Function]}
                  />
                  <CTACard
                    collectionFilterChange={[Function]}
                    deliveryDate={null}
                    isCurrentCollectionRecommendation={false}
                    thematicName={null}
                  />
                </div>,
              ],
              "className": "containerBelowFeatured",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <Connect(RecipeCard)
                      index={2}
                      mobileGridView={false}
                      recipe={Immutable.Map {}}
                      showDetailRecipe={[Function]}
                    />,
                  ],
                  "className": "columnBelowFeatured",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": "2-undefined",
                    "nodeType": "class",
                    "props": Object {
                      "index": 2,
                      "mobileGridView": false,
                      "recipe": Immutable.Map {},
                      "showDetailRecipe": [Function],
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                ],
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    Array [
                      <Connect(RecipeCard)
                        index={3}
                        mobileGridView={false}
                        recipe={Immutable.Map {}}
                        showDetailRecipe={[Function]}
                      />,
                    ],
                    <CTACard
                      collectionFilterChange={[Function]}
                      deliveryDate={null}
                      isCurrentCollectionRecommendation={false}
                      thematicName={null}
                    />,
                  ],
                  "className": "columnBelowFeatured",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": "3-undefined",
                    "nodeType": "class",
                    "props": Object {
                      "index": 3,
                      "mobileGridView": false,
                      "recipe": Immutable.Map {},
                      "showDetailRecipe": [Function],
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "function",
                    "props": Object {
                      "collectionFilterChange": [Function],
                      "deliveryDate": null,
                      "isCurrentCollectionRecommendation": false,
                      "thematicName": null,
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                ],
                "type": "div",
              },
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Connect(RecipeCard)
              index={1}
              mobileGridView={false}
              recipe={Immutable.Map {}}
              showDetailRecipe={[Function]}
            />,
          ],
          "className": "mainColumnRight",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "1-undefined",
            "nodeType": "class",
            "props": Object {
              "index": 1,
              "mobileGridView": false,
              "recipe": Immutable.Map {},
              "showDetailRecipe": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="mainColumnLeft"
          >
            <div
              className="featured"
            >
              <Connect(RecipeCard)
                index={0}
                isFeatured={true}
                mobileGridView={false}
                recipe={Immutable.Map {}}
                showDetailRecipe={[Function]}
              />
            </div>
            <div
              className="containerBelowFeatured"
            >
              <div
                className="columnBelowFeatured"
              >
                <Connect(RecipeCard)
                  index={2}
                  mobileGridView={false}
                  recipe={Immutable.Map {}}
                  showDetailRecipe={[Function]}
                />
              </div>
              <div
                className="columnBelowFeatured"
              >
                <Connect(RecipeCard)
                  index={3}
                  mobileGridView={false}
                  recipe={Immutable.Map {}}
                  showDetailRecipe={[Function]}
                />
                <CTACard
                  collectionFilterChange={[Function]}
                  deliveryDate={null}
                  isCurrentCollectionRecommendation={false}
                  thematicName={null}
                />
              </div>
            </div>
          </div>,
          <div
            className="mainColumnRight"
          >
            <Connect(RecipeCard)
              index={1}
              mobileGridView={false}
              recipe={Immutable.Map {}}
              showDetailRecipe={[Function]}
            />
          </div>,
        ],
        "className": "desktopRecipeList",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="featured"
              >
                <Connect(RecipeCard)
                  index={0}
                  isFeatured={true}
                  mobileGridView={false}
                  recipe={Immutable.Map {}}
                  showDetailRecipe={[Function]}
                />
              </div>,
              <div
                className="containerBelowFeatured"
              >
                <div
                  className="columnBelowFeatured"
                >
                  <Connect(RecipeCard)
                    index={2}
                    mobileGridView={false}
                    recipe={Immutable.Map {}}
                    showDetailRecipe={[Function]}
                  />
                </div>
                <div
                  className="columnBelowFeatured"
                >
                  <Connect(RecipeCard)
                    index={3}
                    mobileGridView={false}
                    recipe={Immutable.Map {}}
                    showDetailRecipe={[Function]}
                  />
                  <CTACard
                    collectionFilterChange={[Function]}
                    deliveryDate={null}
                    isCurrentCollectionRecommendation={false}
                    thematicName={null}
                  />
                </div>
              </div>,
            ],
            "className": "mainColumnLeft",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <Connect(RecipeCard)
                  index={0}
                  isFeatured={true}
                  mobileGridView={false}
                  recipe={Immutable.Map {}}
                  showDetailRecipe={[Function]}
                />,
                "className": "featured",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "index": 0,
                  "isFeatured": true,
                  "mobileGridView": false,
                  "recipe": Immutable.Map {},
                  "showDetailRecipe": [Function],
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <div
                    className="columnBelowFeatured"
                  >
                    <Connect(RecipeCard)
                      index={2}
                      mobileGridView={false}
                      recipe={Immutable.Map {}}
                      showDetailRecipe={[Function]}
                    />
                  </div>,
                  <div
                    className="columnBelowFeatured"
                  >
                    <Connect(RecipeCard)
                      index={3}
                      mobileGridView={false}
                      recipe={Immutable.Map {}}
                      showDetailRecipe={[Function]}
                    />
                    <CTACard
                      collectionFilterChange={[Function]}
                      deliveryDate={null}
                      isCurrentCollectionRecommendation={false}
                      thematicName={null}
                    />
                  </div>,
                ],
                "className": "containerBelowFeatured",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      <Connect(RecipeCard)
                        index={2}
                        mobileGridView={false}
                        recipe={Immutable.Map {}}
                        showDetailRecipe={[Function]}
                      />,
                    ],
                    "className": "columnBelowFeatured",
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": "2-undefined",
                      "nodeType": "class",
                      "props": Object {
                        "index": 2,
                        "mobileGridView": false,
                        "recipe": Immutable.Map {},
                        "showDetailRecipe": [Function],
                      },
                      "ref": null,
                      "rendered": null,
                      "type": [Function],
                    },
                  ],
                  "type": "div",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      Array [
                        <Connect(RecipeCard)
                          index={3}
                          mobileGridView={false}
                          recipe={Immutable.Map {}}
                          showDetailRecipe={[Function]}
                        />,
                      ],
                      <CTACard
                        collectionFilterChange={[Function]}
                        deliveryDate={null}
                        isCurrentCollectionRecommendation={false}
                        thematicName={null}
                      />,
                    ],
                    "className": "columnBelowFeatured",
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": "3-undefined",
                      "nodeType": "class",
                      "props": Object {
                        "index": 3,
                        "mobileGridView": false,
                        "recipe": Immutable.Map {},
                        "showDetailRecipe": [Function],
                      },
                      "ref": null,
                      "rendered": null,
                      "type": [Function],
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "function",
                      "props": Object {
                        "collectionFilterChange": [Function],
                        "deliveryDate": null,
                        "isCurrentCollectionRecommendation": false,
                        "thematicName": null,
                      },
                      "ref": null,
                      "rendered": null,
                      "type": [Function],
                    },
                  ],
                  "type": "div",
                },
              ],
              "type": "div",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Connect(RecipeCard)
                index={1}
                mobileGridView={false}
                recipe={Immutable.Map {}}
                showDetailRecipe={[Function]}
              />,
            ],
            "className": "mainColumnRight",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "1-undefined",
              "nodeType": "class",
              "props": Object {
                "index": 1,
                "mobileGridView": false,
                "recipe": Immutable.Map {},
                "showDetailRecipe": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenAdapter {
      "options": Object {
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "prevContext": true,
          },
          "getChildContext": Object {
            "calledByRenderer": true,
          },
        },
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
